docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -42,6 +42,619 @@ For these cases, SNSs have so called 'generic proposals'. These are custom propo
+        wait_for_quiet_deadline_increase_seconds : opt nat64;
+        max_number_of_neurons : opt nat64;
+        transaction_fee_e8s : opt nat64;
+        max_number_of_proposals_with_ballots : opt nat64;
+        max_age_bonus_percentage : opt nat64;
+        neuron_grantable_permissions : opt record { permissions : vec int32 };
+        voting_rewards_parameters : opt record {
+            final_reward_rate_basis_points : opt nat64;
+            initial_reward_rate_basis_points : opt nat64;
+            reward_rate_transition_duration_seconds : opt nat64;
+            round_duration_seconds : opt nat64;
+        };
+        maturity_modulation_disabled : opt bool;
+        max_number_of_principals_per_neuron : opt nat64;
+    }
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "Lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            ManageNervousSystemParameters: record {
+                default_followees : opt record { 
+                    followees : vec record { 
+                                nat64; 
+                                record { followees : vec record { id : vec {1:nat8, 2:nat8} } }  
+                    } 
+                };
+                max_dissolve_delay_seconds = opt 42:nat64;
+                max_dissolve_delay_bonus_percentage = opt 42:nat64;
+                max_followees_per_function = opt 42:nat64;
+                neuron_claimer_permissions = opt record { permissions = vec {1:int32, 2:int32} };
+                neuron_minimum_stake_e8s = opt 42:nat64;
+                max_neuron_age_for_age_bonus = opt 42:nat64;
+                initial_voting_period_seconds = opt 42:nat64;
+                neuron_minimum_dissolve_delay_to_vote_seconds = opt 42:nat64;
+                reject_cost_e8s = opt 42:nat64;
+                max_proposals_to_keep_per_action = opt 42:nat32;
+                wait_for_quiet_deadline_increase_seconds = opt 42:nat64;
+                max_number_of_neurons = opt 42:nat64;
+                transaction_fee_e8s = opt 42:nat64;
+                max_number_of_proposals_with_ballots = opt 42:nat64;
+                max_age_bonus_percentage = opt 42:nat64;
+                neuron_grantable_permissions = opt record { permissions : vec {1:int32, 2:int32} };
+                voting_rewards_parameters = opt record {
+                    final_reward_rate_basis_points = opt 42:nat64;
+                    initial_reward_rate_basis_points = opt 42:nat64;
+                    reward_rate_transition_duration_seconds = opt 42:nat64;
+                    round_duration_seconds = opt 42:nat64;
+                };
+                maturity_modulation_disabled = opt true:bool;
+                max_number_of_principals_per_neuron = opt 42:nat64;
+            };
+    }
+)'
+```
+
+## `AddGenericNervousSystemFunction`
```

Comment:
I would separate this and the "SubmitGenericNervousSystemFunction" into a new section.

I would actually structure it like this (but we can also do this after example here are clear):
Background
- what are proposals (calls to methods..)
- how to submit a proposal (high level idea of quill, owning a neuron, what you need as prerequisites)

Native proposals
- explained on high level
- list all of them
- for each provide concrete example as now done here, for stuff relevant for generic proposals refer to next section

Generic proposals 
- what they are / how they work, including their structure (namely that they have an id but you cannot use certain ids as they are reserved for native proposals)
- how to add / remove generic proposal, including the example here (this will make the explanation here a lot easier)
- how to submit a generic proposal with example such as here

Reply:
Agreed, I just need some examples of generic proposals. will ask

Reply:
Sent you a concrete example on slack (linked doc)